app/controllers/patients_controller.rb

```diff
@@ -550,6 +549,120 @@ def update_monitoring_fields(patient, params, household, propagation)
     patient.monitoring_history_edit(history_data, diff_state)
   end
 
+  # Update the patient's follow-up flag fields
+  def update_follow_up_flag
+    redirect_to(root_url) && return unless current_user.can_edit_patient_monitoring_info?
+
+    patient = current_user.get_patient(params.permit(:id)[:id])
+    redirect_to(root_url) && return if patient.nil?
+
+    update_follow_up_flag_fields(patient, params)
+  end
+
+  # Make updates to follow-up flag reason and/or note and create corresponding History items.
+  #
+  # patient - The Patient to update.
+  # params - The request params.
+  def update_follow_up_flag_fields(patient, params)
+    clear_flag = params.permit(:clear_flag)[:clear_flag]
+    history_data = {}
+    if clear_flag
+      clear_flag_reason = params.permit(:clear_flag_reason)[:clear_flag_reason]
+      clear_follow_up_flag(patient, clear_flag_reason)
+    else
+      follow_up_reason = params.permit(:follow_up_reason)[:follow_up_reason]
+      follow_up_note = params.permit(:follow_up_note)[:follow_up_note]
+
+      # Handle creating history items based on the updates
+      history_data = {
+        created_by: current_user.email,
+        patient: patient,
+        follow_up_reason: follow_up_reason,
+        follow_up_note: follow_up_note,
+        follow_up_reason_before: patient.follow_up_reason,
+        follow_up_note_before: patient.follow_up_note
+      }
+
+      # Handle success or failure of updating a follow-up flag
+      ActiveRecord::Base.transaction do
+        # Apply and save updates to the db
+        if patient.update!(follow_up_reason: follow_up_reason, follow_up_note: follow_up_note)
+          # Create history item on successful update
+          History.follow_up_flag_edit(history_data)
+        else
+          # Handle case where follow-up flag update failed
+          error_message = 'Unable to update Follow-up Flag.'
+          render(json: { error: error_message }, status: :bad_request) && return
+        end
```

Comment:
The `else` case will never be hit, because `update!` will throw an error if it fails. We do want an error to be thrown so the transaction rolls back. I think it would be best to just remove the `else` case.